src/test/java/com/cloudbees/jenkins/support/impl/RunningJobsTest.java

```diff
@@ -0,0 +1,45 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.SupportTestUtils;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.queue.QueueTaskFuture;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertThat;
+
+public class RunningJobsTest {
+
+    private static final String JOB_NAME = "job-name";
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void addContents() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject(JOB_NAME);
+        QueueTaskFuture<FreeStyleBuild> build = p.scheduleBuild2(0);
+        build.waitForStart();
+
+        String output = SupportTestUtils.invokeComponentToString(new RunningJobs());
+
+        assertThat(output, containsString(p.getName()));
+    }
+
+    @Test
+    public void addContentsPipeline() throws Exception {
+        WorkflowJob p = j.createProject(WorkflowJob.class, JOB_NAME);
```

Comment:
I think you are going to need an actual script here! Or the build will just be some degenerate broken thing. Surprised this test can even pass.

```java
p.setDefinition(new CpsFlowDefinition("node {semaphore 'wait'}", true);
WorkflowRun build = p.scheduleBuild2(0).waitForStart();
SemaphoreStep.waitForStart("wait/1", build);
```

and then you want to assert that the output does contain a representation of the build but does _not_ contain a representation of the `node` block.